### PR TITLE
[5.1] Automatically increase artisan queue:listen --timeout value if it's <= --sleep value to prevent unexpected timeout

### LINKED ISSUE #6206: queue:listen's sleep duration contributing towards timeout
To reproduce, create a simple job and call artisan queue listen:

`php artisan queue:listen --sleep=5 --timeout=5`

If you look in your laravel.log, you'll notice that the spawned queue:work processes are all instantly timing out:

```
[2014-10-23 16:17:15] dev.ERROR: exception 'Symfony\Component\Process\Exception\ProcessTimedOutException' with message 'The process ""/usr/bin/php5" artisan queue:work  --queue="default" --delay=0 --memory=128 --sleep=5 --tries=0 --env=dev" exceeded the timeout of 5 seconds.' in /.../vendor/symfony/process/Symfony/Component/Process/Process.php:1209
Stack trace:
#0 /home/.../vendor/symfony/process/Symfony/Component/Process/Process.php(357): Symfony\Component\Process\Process->checkTimeout()
#1 /home/.../vendor/symfony/process/Symfony/Component/Process/Process.php(206): Symfony\Component\Process\Process->wait()
#2 /home/.../vendor/laravel/framework/src/Illuminate/Queue/Listener.php(94): Symfony\Component\Process\Process->run(Object(Closure))
...
```

If you change the listen's timeout to 6 seconds, the jobs will run successfully:

`php artisan queue:listen --sleep=5 --timeout=6`

From my perspective (and after reading the [queue docs](http://laravel.com/docs/4.2/queues)), it seems like the specified sleep duration should not contribute towards the timeout duration. The listen process should wait \<sleep\> seconds before checking if there's a job to run. If there is a job, the job process is spawned and expires in \<timeout\> seconds.

This is happening in Laravel 4.2.11.

### PR DESCRIPTION
**Pull request by jairovsky**
Fixes #6206.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Please just update to throw an InvalidArgumentException.

**Comment by GrahamCampbell** [comment]
You can do this in place if you want.

File: src/Illuminate/Queue/Console/ListenCommand.php [hunk 1]
```diff
@@ -63,6 +63,12 @@ public function fire()
 
         $timeout = $this->input->getOption('timeout');
 
+        $sleep = $this->input->getOption('sleep');
```
Comment (lucasmichot) [inline src/Illuminate/Queue/Console/ListenCommand.php]: Why not just:
```php
if ($timeout <= $this->input->getOption('sleep')) {
```
as `$sleep` is only used once

File: src/Illuminate/Queue/Console/ListenCommand.php [hunk 2]
```diff
@@ -63,6 +63,12 @@ public function fire()
...
+
+        if ($timeout <= $sleep) {
+            throw new \InvalidArgumentException("--timeout value must be higher than --sleep");
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/Console/ListenCommand.php]: Please import this class.

Comment (lucasmichot) [inline src/Illuminate/Queue/Console/ListenCommand.php]: And add @throws \InvalidArgumentException to the function docblock

Comment (GrahamCampbell) [inline src/Illuminate/Queue/Console/ListenCommand.php]: Also, please provide a better exception message.

Comment (GrahamCampbell) [inline src/Illuminate/Queue/Console/ListenCommand.php]: `throw new InvalidArgumentException('The timeout must be greater than the sleep time.');`

**Comment by taylorotwell** [comment]
Fixed.

**Comment by yooper** [comment]
When the timeout is set to zero it allows for the job to run for an indefinite amount of time. We have very long running jobs that vary in processing time based on the volume of data. This change forces us to set a timeout value, otherwise an exception is thrown, because timeout must **now** be greater than the sleep value. Please remove this restriction because it causes unintended consequences for people running large data processing jobs. It causes more harm than good.

**Comment by viraj-khatavkar** [comment]
@taylorotwell Please remove this restriction because it is very much specific to a particular usecase and as mentioned by @yooper causes unintended consequences for people running large data processing jobs. It causes more harm than good.

**Comment by taylorotwell** [comment]
Reverted on 5.1.

**Comment by viraj-khatavkar** [comment]
@taylorotwell I still get an error `Job timeout must be greater than 'sleep' option value.` if I fire `php artisan queue:listen --timeout=0`

I am on `5.1.27`

**Comment by GrahamCampbell** [comment]
It's not available in a tagged release yet.

**Comment by viraj-khatavkar** [comment]
@GrahamCampbell @taylorotwell is this going to be tagged on 5.2 and 5.1 as well ? Cos I am on verge of upgrading to 5.2

**Comment by GrahamCampbell** [comment]
It's available in all future 5.1 and 5.2 releases.

**Comment by viraj-khatavkar** [comment]
@GrahamCampbell - When will this change be available in tagged release of 5.2 ?

**Pull request closed without merging**